testsuite/testcases/src/state_sync_performance.rs

```diff
@@ -52,70 +73,154 @@ impl NetworkTest for StateSyncPerformance {
-                "State sync TPS requirement failed. Average TPS: {}, minimum required TPS: {}",
-                state_sync_throughput, min_expected_tps
-            );
-            bail!(error_message)
+        // Generate some traffic through the validators.
+        // We do this for half the test time.
+        let emit_txn_duration = ctx.global_duration.checked_div(2).unwrap();
+        info!(
+            "Generating the initial traffic for {:?} seconds.",
+            emit_txn_duration.as_secs()
+        );
+        let _txn_stat = generate_traffic(ctx, &all_validators, emit_txn_duration, 1)?;
+
+        // Wait for all nodes to synchronize and stabilize.
+        info!("Waiting for the validators to be synchronized.");
+        let runtime = Runtime::new().unwrap();
+        runtime.block_on(async {
+            ctx.swarm()
+                .wait_for_all_nodes_to_catchup(Duration::from_secs(MAX_NODE_LAG_SECS))
+                .await
+        })?;
+
+        // Stop and reset two validators
+        info!("Deleting data for two validators!");
+        let validators_to_restart = &all_validators[0..2];
+        for validator_id in validators_to_restart {
+            let validator = ctx.swarm().validator_mut(*validator_id).unwrap();
+            runtime.block_on(async { validator.clear_storage().await })?;
+        }
+
+        // Restart the validators so they start syncing from a fresh state
+        // and start the timer.
+        for validator_id in validators_to_restart {
+            let validator = ctx.swarm().validator_mut(*validator_id).unwrap();
+            runtime.block_on(async { validator.start().await })?;
         }
+        let timer = Instant::now();
+
+        // Wait for all nodes to catch up to the highest synced version
+        // then calculate and display the throughput results.
+        ensure_state_sync_throughput(ctx, timer, self.name())
+    }
+}
 
-        Ok(())
+/// Calculates, enforces and displays the state sync throughput using
+/// the synced version and sync duration.
+fn ensure_state_sync_throughput(
+    ctx: &mut NetworkContext<'_>,
+    timer: Instant,
+    test_name: &str,
+) -> Result<()> {
+    // Get the highest synced version for the chain
+    let runtime = Runtime::new().unwrap();
+    let highest_synced_version = runtime.block_on(async {
+        get_highest_synced_version(&ctx.swarm().get_all_nodes_clients_with_names())
+            .await
+            .unwrap_or(0)
+    });
+    if highest_synced_version == 0 {
+        return Err(anyhow::format_err!(
+            "The swarm has synced 0 versions! Something has gone wrong!"
+        ));
     }
+
+    // Wait for all nodes to catch up to the same synced version.
+    // We allow up to half the test time to do this.
+    let node_sync_duration = ctx.global_duration.checked_div(2).unwrap();
+    runtime.block_on(async {
+        ctx.swarm()
+            .wait_for_all_nodes_to_catchup(node_sync_duration)
```

Comment:
you can use wait_for_all_nodes_to_catchup_to_version with highest_synced_version, if you want (to make metric ever so slightly more precise :) )